src/hotspot/cpu/s390/templateInterpreterGenerator_s390.cpp

```diff
@@ -2006,8 +2006,19 @@ address TemplateInterpreterGenerator::generate_CRC32C_updateBytes_entry(Abstract
   return __ addr_at(entry_off);
 }
 
+address TemplateInterpreterGenerator::generate_currentThread() {
+  address entry_point = __ pc();
```

Comment:
I would suggest to remember the entry offset, not the entry address, just like in the generator above.
In general, it is possible that, while generating code, the allocated space gets exhausted. In that case, a new buffer is allocated and the code generated so far is copied to the new space. All remembered addresses become invalid then. Offsets are relative to the begin of the buffer and remain valid. 

Afaik, this concern is not relevant when generating the interpreter. Here, the allocated space must be large enough from the beginning. Doing it right anyway would be nice. Thanks.

Reply:
I have updated the code. Thanks for the explanation as well :)